### PR TITLE
adding sdn workflows for aws workers-rhel8

### PR DESCRIPTION
**Pull request by lihongan**

### CODE REVIEW

**Comment by lihongan** [comment]
```
"installer-rehearse-aws" pod "installer-rehearse-aws-workers-rhel-pre-hook-sshtunnel" failed: the pod ci-op-tvbc3ccd/installer-rehearse-aws-workers-rhel-pre-hook-sshtunnel failed after 2h0m14s (failed containers: test): ContainerFailed one or more containers exited
```

**Comment by lihongan** [comment]
Looks timeout again for step `workers-rhel-pre-hook-sshtunnel`, see https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31307/rehearse-31307-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1559128960380440576#1:build-log.txt%3A12

Try to check workflow cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn and re-trigger the test

**Comment by jianlinliu** [comment]
> Looks timeout again for step `workers-rhel-pre-hook-sshtunnel`, see https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31307/rehearse-31307-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1559128960380440576#1:build-log.txt%3A12
> 
> Try to check workflow cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn and re-trigger the test

Sounds like there is script bug on ssh-tunnel, we are trying to fix it.

**Comment by lihongan** [comment]
> > Looks timeout again for step `workers-rhel-pre-hook-sshtunnel`, see https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31307/rehearse-31307-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1559128960380440576#1:build-log.txt%3A12
> > Try to check workflow cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn and re-trigger the test
> 
> Sounds like there is script bug on ssh-tunnel, we are trying to fix it.

Yes, looks like something wrong with ssh-tunnel, the job for workflow aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn also hit the same issue.

**Comment by jianlinliu** [comment]
let us hold testing for a while, once the issue is fixed, will notice you, after you can you sync upstream to retest it.

**Comment by jianlinliu** [comment]
@lihongan https://github.com/openshift/release/pull/31362 should fix the ssh-tunnel issue, pls sync upstream, have a retest.

**Comment by lihongan** [comment]
Thank you @jianlinliu , let me retest workflow `cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn` firstly.

**Comment by lihongan** [comment]
```
Step installer-rehearse-aws-workers-rhel-repos failed
```
/retest

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-workflow.metadata.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+	"path": "cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-workflow.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-workflow.metadata.json:2]: How about move the `sdn` folder to `aws/ipi/proxy`?

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+	"path": "cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json:2]: How about move `sdn` folder to `aws/ipi/disconnected/private`, then we can have `cucushift/installer/rehearse/aws/ipi/disconnected/private/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-sdn/-rovision-chain.yaml`, then we can reference the provision chain in the rhel workflow.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-provision-chain.metadata.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+	"path": "cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-provision-chain.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-provision-chain.metadata.json:2]: How about moving sdn to `aws/ipi/proxy/`, then we will have cucushift-installer-rehearse-aws-ipi-proxy-sdn-provision-chain.yaml, based that chain, we can add rhel-workers steps to have `cucushift-installer-rehearse-aws-ipi-proxy-sdn-workers-rhel8-provision-chain.yaml`

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json (hunk 2 above)
Comment (lihongan) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json:2]: Good idea. 
Sure, I can have a try.

**Comment by jianlinliu** [comment]
For the downloading oc issue, I saw several recent jobs also hit such issue, sounds like test farm cluster network issues to target server:
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.12-e2e-aws-workers-rhel8/1560058909832515584
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.11-e2e-aws-workers-rhel8/1560052618019999744
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.10-e2e-aws-workers-rhel8/1560095774010773504

I will try to figure out some enhancement in another PR.

**Comment by jianlinliu** [comment]
https://github.com/openshift/release/pull/31256 is trying to fix oc binary downloading issue, once that PR get merge, you can sync and retest.

**Comment by jianlinliu** [comment]
@lihongan https://github.com/openshift/release/pull/31256 get merged, pls retry

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json (hunk 2 above)
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json:2]: Though `cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/` and `cucushift/installer/rehearse/aws/ipi/disconnected/private/sdn/workers-rhcos-rhel8/provision/` should have the same effect, for keeping consistent folder layout, do you mind update the layout a bit to `cucushift/installer/rehearse/aws/ipi/disconnected/private/sdn/workers-rhcos-rhel8/provision/`

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-provision-chain.metadata.json (hunk 3 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/workers-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-proxy-workers-rhel8-sdn-provision-chain.metadata.json:2]: Similar suggestion to https://github.com/openshift/release/pull/31307/files#r954923552

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json (hunk 2 above)
Comment (lihongan) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/sdn/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-sdn-provision-chain.metadata.json:2]: Sure, will update it.

**Comment by lihongan** [comment]
/hold
since I need to test again for another workflow

**Comment by lihongan** [comment]
The test for `cucushift-installer-rehearse-aws-ipi-proxy-sdn-workers-rhel8` looks good, let's test another one.

**Pull request merged**